### PR TITLE
Issue1013, New API: get transaction hex without `signatures` field

### LINKED ISSUE #1013: New API: get transaction hex without `signatures` field
**User Story**
As a node API user I want features:
* get hex of a signed transaction (which is current behavior of `get_transaction_hex` API, if there is no signature in the transaction, there will be a byte with value `00` at the end of the hex);
* get hex of a transaction without signatures (actually without the `signature` field), so I can sign it directly.

**Additional Context**
I've heard of complaints from different people about the extra `00` byte returned by `get_transaction_hex` API. Not sure how that API is being used by client software, so IMHO it's safer to keep its behavior and add a new API (although don't know whether/how the new API will be used as well).

## CORE TEAM TASK LIST
- [x] Evaluate / Prioritize Feature Request
- [x] Refine User Stories / Requirements
- [ ] Define Test Cases
- [x] Design / Develop Solution
- [ ] Perform QA/Testing
- [ ] Update Documentation

### PR DESCRIPTION
**Pull request by cifer-lee**
fix #1013

### CODE REVIEW

File: libraries/app/include/graphene/app/database_api.hpp [hunk 1]
```diff
@@ -596,6 +596,11 @@ class database_api
       /// @brief Get a hexdump of the serialized binary form of a transaction
       std::string get_transaction_hex(const signed_transaction& trx)const;
 
+      /// @brief Get a hexdump of the serialized binary form of a
+      /// signatures-stripped transaction
+      std::string
+      get_transaction_hex_without_sig(const signed_transaction &trx) const;
+
```
Comment (oxarbitrage) [inline libraries/app/include/graphene/app/database_api.hpp]: please put `std::string` and `get_transaction_hex_without_sig(const signed_transaction &trx) const;` in the same line. thanks!

Comment (oxarbitrage) [reply libraries/app/include/graphene/app/database_api.hpp]: i am wondering also what will be happening if a transaction with signatures came here. what will be the output ? maybe add a test case for that?

Comment (pmconrad) [reply libraries/app/include/graphene/app/database_api.hpp]: +1 please add a test case.

Comment (cifer-lee) [reply libraries/app/include/graphene/app/database_api.hpp]: Thanks Alfredo and Peter, fixed

**Pull request merged**